docs/src/extend/code-path-analysis.md

```diff
@@ -272,23 +298,28 @@ function isCbCalled(info) {
 }
 
 module.exports = function(context) {
-    var funcInfoStack = [];
-    var segmentInfoMap = Object.create(null);
+    let funcInfo;
+    const funcInfoStack = [];
+    const segmentInfoMap = Object.create(null);
 
     return {
         // Checks `cb`.
-        "onCodePathStart": function(codePath, node) {
-            funcInfoStack.push({
+        onCodePathStart(codePath, node) {
+            funcInfo = {
                 codePath: codePath,
-                hasCb: hasCb(node, context)
-            });
+                hasCb: hasCb(node, context),
+                currentSegments: []
+            };
+
+            funcInfoStack.push(funcInfo);
         },
-        "onCodePathEnd": function(codePath, node) {
-            funcInfoStack.pop();
+
+        onCodePathEnd(codePath, node) {
+            funcInfo = funcInfoStack.pop();
```

Comment:
After `onCodePathEnd`, `funcInfo` remains the same as it's the last one on the stack, which I think wasn't intended.

Reply:
Looks like I need to revisit this page as the code examples are all pretty old an ES5-style.